src/query/query_executor.c

```diff
@@ -23807,7 +23808,8 @@ qexec_evaluate_aggregates_optimize (THREAD_ENTRY * thread_p, AGGREGATE_TYPE * ag
     {
       if (agg_ptr->flag_agg_optimize)
 	{
-	  if (agg_ptr->function == PT_COUNT_STAR && *is_scan_needed)
+	  if ((agg_ptr->function == PT_COUNT_STAR || agg_ptr->function == PT_MAX || agg_ptr->function == PT_MIN)
+	      && *is_scan_needed)
```

Comment:
This code does not exist in version 9. What is it for?

Reply:
I added the code to fix some sql fail cases, but I missed to revert it.